core/io/resource.cpp

```diff
@@ -222,6 +291,45 @@ Ref<Resource> Resource::duplicate_for_local_scene(Node *p_for_scene, HashMap<Ref
 	return r;
 }
 
+static void _find_sub_resources(const Variant &v, HashSet<Ref<Resource>> &resources_found) {
+	switch (v.get_type()) {
+		case Variant::ARRAY: {
+			Array a = v;
+			for (int i = 0; i < a.size(); i++) {
+				Variant v2 = a[i];
+				if (v2.get_type() == Variant::ARRAY || v2.get_type() == Variant::DICTIONARY || v2.get_type() == Variant::OBJECT) {
```

Comment:
This check is already done by the `switch` in `_find_sub_resources`, so it's not needed here.